app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -262,6 +269,37 @@ def remove_disk_config_spec(vim_obj, vmcs, hardware, options)
     end
   end
 
+  def resize_disk_config_spec(vim_obj, vmcs, hardware, options)
+    raise "resize_disk_config_spec: disk filename is required." unless options[:disk_name]
+
+    controller_key, key = vim_obj.getDeviceKeysByBacking(options[:disk_name], hardware)
+    raise "resize_disk_config_spec: no virtual device associated with: #{options[:disk_name]}" unless key
+
+    device = vim_obj.getDeviceByBacking(options[:disk_name])
```

Comment:
Ah here's an example of where we can pass in the hardware hash and save an API call.  If you call just this method you can get the controller_key and key from the device returned.

Reply:
Yup I will fix that